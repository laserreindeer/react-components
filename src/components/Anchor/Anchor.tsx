import { wrap } from "../../utils/wrap/wrap";
import type { AnchorProperties } from "./AnchorProperties";

/**
 * Anchor component safe by default and with aside functionality.
 */
export const Anchor = wrap("a")<AnchorProperties>(
	({ aside = false, ...properties }) => (
		<a
			rel="noopener noreferrer"
			target={aside ? "_blank" : undefined}
			{...properties}
		/>
	)
);
