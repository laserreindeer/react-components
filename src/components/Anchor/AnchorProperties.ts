/**
 * Anchor component properties.
 */
export type AnchorProperties = {
	/** Open in a new window */
	aside: boolean;
};
