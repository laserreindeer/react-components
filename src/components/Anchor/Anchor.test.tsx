import { render } from "@testing-library/react";
import { Anchor } from "./Anchor";

it("matches snapshot", () => {
	const { asFragment } = render(<Anchor>Test</Anchor>);
	expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot aside", () => {
	const { asFragment } = render(<Anchor aside>Test</Anchor>);
	expect(asFragment()).toMatchSnapshot();
});
