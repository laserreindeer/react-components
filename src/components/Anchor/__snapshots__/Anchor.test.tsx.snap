// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    rel="noopener noreferrer"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`matches snapshot aside 1`] = `
<DocumentFragment>
  <a
    rel="noopener noreferrer"
    target="_blank"
  >
    Test
  </a>
</DocumentFragment>
`;
